keps/993-two-phase-admission/README.md

```diff
@@ -253,6 +254,40 @@ The controller implementing a particular check should:
 * After approving the workload, keep an eye on the check if it starts failing,
   fail the check and cause the workload to move back to the suspended state.
 
+### Preemption Admission Check Controller
+
+In this proposal, the time to evict the preemption candidates varies based on the preemptor state
+the scheduler will not issue the eviction during it's process instead it will set a `Pending` admission check
+that is manged by a new built-in admission check controller.
+
+The **Preemption Admission Check Controller** will:
+
+- Watch for a change in state of the workloads pending preemption.
+- Watch for workloads that are finishing execution and therefore releasing quota.
+- Watch for AdmissionCheck changes, since the preemption policy can change.
+
+The preemption controller uses the kueue cache, since it needs to check the state of workloads admitted to the ClusterQueues.
+
+At every run the controller will get the list of workloads pending preemption.
+
+The workloads pending preemption are divided into:
+- `preemtingLetaer` - Workloads having at least one check that uses AfterCheckPassedOrOnDemand policy with the state `pending`.
+- `preemtingNow` - Workloads that expect to be able to issue evictions or potentially change their preemption state in the current cycle.
+
+Then:
+1. Remove all workloads from the snapshot.
+2. For every workload in `preemtingNow` , in the order of their priority:
+  - If it can fit without the need of evicting other workloads
+    - Set its admission check to `Ready`
+    - Add it to the snapshot
+  - If it cannot fit
+    - Get an updated list of eviction candidates
+      - If the updated list is not empty.
+        - Issue the eviction to the candidates.
+        - Add it to the snapshot
```

Comment:
At this point? Shouldn't we wait for the next reconcile, given that jobs can take time to terminate?

Reply:
Just to be clear, you mean the snapshot, not the cache, correct?

Reply:
Shapshot